### PR TITLE
MOD-8050 MOD-8052: Add ACL username config param

### PR DESCRIPTION
**Pull request by raz-mon**
**Describe the changes in the pull request**

We add a new load-time configuration parameter `OSS_ACL_USERNAME`, which is the username with which the coordinator authenticates against the shard connections it holds.
This, together with the existing `OSS_GLOBAL_PASSWORD` will allow the user to configure an ACL user that will be able to execute RediSearch internal commands.

* Notice: Since we are talking about load-time config-params, once a user connects with a bad username\password, this will trigger an authentication loop in our uv-thread, that will end only when the config-params' values are set to the correct values, and the shards are reset (i.e., shutdown and started back). This is the current state we have with the existing password config-param as well. A change of that mechanism can be discussed outside the scope of this PR.

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/coord/rmr/conn.c
Comment (kei-nan) [inline src/coord/rmr/conn.c:9]: can you fix this include line?
I don't think we generally use relative path includes in our code.

File: src/coord/rmr/conn.c
Comment (kei-nan) [inline src/coord/rmr/conn.c:430]: I don't get it, looked at all the `MRConn_SendAuth` calls.
It seems like if it returns failure then we change the state to connecting, which is fine.
But internally in between we change the state to Reauth, which seems to start a timer.
Isn't there a better way of doing this?
Seems redundant to move to reauth state just to immediately move to connecting state.

File: tests/pytests/test_config.py
Comment (kei-nan) [inline tests/pytests/test_config.py:7]:
```suggestion
def _test_config_str(arg_name, arg_value, ret_value=None):
```

File: tests/pytests/test_config.py
Comment (kei-nan) [inline tests/pytests/test_config.py:16]:
```suggestion
def _test_config_num(arg_name, arg_value):
```

File: tests/pytests/test_config.py
Comment (kei-nan) [inline tests/pytests/test_config.py:23]:
```suggestion
def _test_config_boolean(arg_name, res):
```

File: tests/pytests/test_config.py
Comment (kei-nan) [inline tests/pytests/test_config.py:409]:
```suggestion
    timeout = 3 # 3 seconds, more than enough for the an env to be up
```

File: tests/pytests/test_config.py
Comment (kei-nan) [inline tests/pytests/test_config.py:411]: I  think the behavior should be that the user gets a proper error.
I don't think we should be in an infinite loop.

File: tests/pytests/test_config.py
Comment (kei-nan) [inline tests/pytests/test_config.py:415]: I think tests that rely on timeouts tend to eventually become flaky.

**Review by kei-nan — CHANGES_REQUESTED** [review]
Left a few comments, 
Mainly I think the infinite loop scenario is not ideal and I suggest considering returning a proper error to the end user.

File: src/coord/rmr/conn.c
Comment (raz-mon) [reply src/coord/rmr/conn.c:430]: Where do we change to connecting?
I think we stay on `ReAuth`.

File: tests/pytests/test_config.py
Comment (raz-mon) [reply tests/pytests/test_config.py:411]: I agree in general, but this is an interesting state cluster-wise as well, thus should be thought about separately since this is an existing issue.
Currently the cluster will need a restart for the new load-time config-param values anyways so I'm not sure this is a big problem.
Bottom line - this is the current behavior, and if we want to fix it we need to think about it more and do it in a more complete manner separately.

File: tests/pytests/test_config.py
Comment (raz-mon) [reply tests/pytests/test_config.py:415]: I agree it's not ideal, yet a hanging client is the current behavior of this situation, so until we don't return a proper error for this situation I have no other way to test this.
I can comment this part out for now if we don't want it to be part of our CI (that I can understand and agree with)

File: tests/pytests/test_config.py
Comment (kei-nan) [reply tests/pytests/test_config.py:411]: I would accept the existing behavior argument if the test didn't exist.
The fact you created a test to verify this behavior seems to imply this behavior is part of the code you added.
Otherwise what exactly did you add behavior wise that you want to test.

File: src/coord/rmr/conn.c
Comment (kei-nan) [reply src/coord/rmr/conn.c:430]: See all the places `MRConn_SendAuth` gets called and what happens when it returns an error code to the caller.
The calling code proceeds to then change the start to connecting.
Unless I'm missing something.

File: tests/pytests/test_config.py
Comment (oshadmi) [inline tests/pytests/test_config.py:176]: Remove line?

File: src/coord/rmr/conn.c
Comment (kei-nan) [reply src/coord/rmr/conn.c:430]: After discussing, we decided it is out of scope.

**Review by kei-nan — APPROVED** [review]

File: src/coord/config.c
Comment (oshadmi) [inline src/coord/config.c:152]: Is this a breaking change?
Is it only breaking the ability to set the password, but actually it was only considered in module init?

File: src/coord/config.c
Comment (oshadmi) [inline src/coord/config.c:173]: Can we do it only in OSS? (use `#ifdef RS_CLUSTER_ENTERPRISE`)

File: src/coord/config.c
Comment (raz-mon) [reply src/coord/config.c:152]: Effectively not, since we rely on load-time setting of this config param only

**Review by oshadmi — APPROVED** [review]

File: src/coord/config.c
Comment (raz-mon) [reply src/coord/config.c:173]: We didn't do this until now, we would need more approvals to start doing it

**Pull request merged**
